### PR TITLE
Update GitHub Actions to avoid `set-output` deprecation

### PR DESCRIPTION
**Pull request by shuuji3**
Starting 1st June 2023 (planned) workflows, `::set-output name...::` syntax cannot be used. You can see the example warnings here: https://github.com/tensorflow/tensorflow/actions/runs/3752114734

ref. [GitHub Actions: Deprecating save-state and set-output commands | GitHub Changelog](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)

This PR fixes the issue by using the `GITHUB_OUTPUT` environment variable as instructed in the above article.

### CODE REVIEW

**Comment by shuuji3** [comment]
Here's the trimmed version of the GitHub Action test run for `.github/workflows/sigbuild-docker-presubmit.yml`: https://github.com/shuuji3/tensorflow/actions/runs/3792737075/jobs/6449190102

It checks only the `DATE` output but confirmed the output is stored as expected. Other outputs would be the same.

File: .github/workflows/release-branch-cherrypick.yml [hunk 1]
```diff
@@ -52,8 +52,8 @@ jobs:
           git config --global user.email "jenkins@tensorflow.org"
           git fetch origin master
           git cherry-pick ${{ github.event.inputs.git_commit }}
-          echo ::set-output name=SHORTSHA::$(git log -1 ${{ github.event.inputs.git_commit }} --format="%h")
-          echo ::set-output name=TITLE::$(git log -1 ${{ github.event.inputs.git_commit }} --format="%s")
+          echo SHORTSHA=$(git log -1 ${{ github.event.inputs.git_commit }} --format="%h") >> $GITHUB_OUTPUT
+          echo TITLE=$(git log -1 ${{ github.event.inputs.git_commit }} --format="%s") >> $GITHUB_OUTPUT
```
Comment (mihaimaruseac) [inline .github/workflows/release-branch-cherrypick.yml:56]:
```suggestion
          echo "SHORTSHA=$(git log -1 ${{ github.event.inputs.git_commit }} --format="%h")" >> "$GITHUB_OUTPUT"
          echo "TITLE=$(git log -1 ${{ github.event.inputs.git_commit }} --format="%s")" >> "$GITHUB_OUTPUT"
```

Comment (mihaimaruseac) [inline .github/workflows/release-branch-cherrypick.yml:56]:
```suggestion
          echo "SHORTSHA=$(git log -1 ${{ github.event.inputs.git_commit }} --format="%h")" >> "$GITHUB_OUTPUT"
          echo "TITLE=$(git log -1 ${{ github.event.inputs.git_commit }} --format="%s")" >> "$GITHUB_OUTPUT"
```

File: .github/workflows/sigbuild-docker-branch.yml [hunk 2]
```diff
@@ -55,9 +55,9 @@ jobs:
       -
         name: Generate variables for cache busting and tag naming
         run: |
-          echo "::set-output name=DATE::$(date +'%Y-%m-%d')"
+          echo "DATE=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
```
Comment (mihaimaruseac) [inline .github/workflows/sigbuild-docker-branch.yml:58]:
```suggestion
          echo "DATE=$(date +'%Y-%m-%d')" >> "$GITHUB_OUTPUT"
```

File: .github/workflows/sigbuild-docker-branch.yml [hunk 3]
```diff
@@ -55,9 +55,9 @@ jobs:
...
           # Converts r2.9 to just 2.9
-          echo "::set-output name=REF::$(echo $GITHUB_REF_NAME | sed 's/r//g')"
+          echo "REF=$(echo $GITHUB_REF_NAME | sed 's/r//g')" >> $GITHUB_OUTPUT
```
Comment (mihaimaruseac) [inline .github/workflows/sigbuild-docker-branch.yml:60]:
```suggestion
          echo "REF=$(echo $GITHUB_REF_NAME | sed 's/r//g')" >> "$GITHUB_OUTPUT"
```

Comment (mihaimaruseac) [inline .github/workflows/sigbuild-docker-branch.yml:60]:
```suggestion
          echo "REF=$(echo $GITHUB_REF_NAME | sed 's/r//g')" >> "$GITHUB_OUTPUT"
```

File: .github/workflows/sigbuild-docker-presubmit.yml [hunk 4]
```diff
@@ -48,7 +48,7 @@ jobs:
       -
         name: Grab the date to do cache busting (assumes same day OK to keep)
         run: |
-          echo "::set-output name=DATE::$(date +'%Y-%m-%d')"
+          echo "DATE=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
```
Comment (mihaimaruseac) [inline .github/workflows/sigbuild-docker-presubmit.yml:51]:
```suggestion
          echo "DATE=$(date +'%Y-%m-%d')" >> "$GITHUB_OUTPUT"
```

File: .github/workflows/sigbuild-docker-branch.yml (hunk 3 above)
Comment (mihaimaruseac) [inline .github/workflows/sigbuild-docker-branch.yml:60]:
```suggestion
          echo "REF=$(echo $GITHUB_REF_NAME | sed 's/r//g')" >> "$GITHUB_OUTPUT"
```

File: .github/workflows/sigbuild-docker.yml [hunk 5]
```diff
@@ -61,10 +61,10 @@ jobs:
           # [[:digit:]] searches for numbers and \+ joins them together
           major_version=$(grep "^#define TF_MAJOR_VERSION" ./tensorflow/core/public/version.h | grep -o "[[:digit:]]\+")
           minor_version=$(grep "^#define TF_MINOR_VERSION" ./tensorflow/core/public/version.h | grep -o "[[:digit:]]\+")
-          echo ::set-output name=TF_VERSION::${major_version}.${minor_version}
+          echo TF_VERSION=${major_version}.${minor_version} >> $GITHUB_OUTPUT
```
Comment (mihaimaruseac) [inline .github/workflows/sigbuild-docker.yml:64]:
```suggestion
          echo "TF_VERSION=${major_version}.${minor_version}" >> "$GITHUB_OUTPUT"
```

File: .github/workflows/sigbuild-docker.yml [hunk 6]
```diff
@@ -61,10 +61,10 @@ jobs:
...
           # Also get the current date to do cache busting. Assumes one day
           # is an ok range for rebuilds
-          echo "::set-output name=DATE::$(date +'%Y-%m-%d')"
+          echo "DATE=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
```
Comment (mihaimaruseac) [inline .github/workflows/sigbuild-docker.yml:67]:
```suggestion
          echo "DATE=$(date +'%Y-%m-%d')" >> "$GITHUB_OUTPUT"
```

Comment (mihaimaruseac) [inline .github/workflows/sigbuild-docker.yml:67]:
```suggestion
          echo "DATE=$(date +'%Y-%m-%d')" >> "$GITHUB_OUTPUT"
```

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Review by mihaimaruseac — APPROVED** [review]
Thank you very much!

**Comment by shuuji3** [comment]
Thank you for reviewing and additional improvements! 🙂

**Pull request merged**